### ISSUE TITLE
utils.guessrangefstr incorrectly evaluates for locale '%Y%m%d'

### ISSUE THREAD

**Issue by jakerobers** [pimutils/khal#871]
When dealing with dates in November or December the result of `utils.guessrangefstr` evaluates to a past January 1st date. (Note: this bug only occurs for these months. Probably something to do with the month numeral being 11 and 12).

Locale section of config file:

```
[locale]
timeformat = %H%M
dateformat = %Y%m%d
longdateformat = %Y%m%d
datetimeformat = %Y%m%d%H%M
longdatetimeformat = %Y%m%d%H%M
default_timezone = America/Chicago
```

Confirming locales:
```
$ khal printformats
longdatetimeformat: 201312211009
datetimeformat: 201312211009
longdateformat: 20131221
dateformat: 20131221
timeformat: 1009
```

The bug can be seen when running:

```
khal list 20191128   # some date in november or december. It spits back the results for January 1st
```

I did not trace any further than  `utils.guessrangefstr`. The actual bug could be further down into `guessdatetimefstr` or something,

**Comment by jakerobers** [comment]
It looks like moving https://github.com/pimutils/khal/blob/master/khal/parse_datetime.py#L225 to be the first in the array fixes the issue.

I am unsure if moving this line will introduce other regressions.

**Comment by jakerobers** [comment]
Looks like this does introduce regression issues when running the unit tests. I see why we need to prioritize datetime over a date.

on https://github.com/pimutils/khal/blob/master/khal/parse_datetime.py#L77
```
strptime(dtstring, dateformat)
```

is evaluating to

```
strptime('20191128', '%Y%m%d%H%M')
```

**Comment by geier** [comment]
I agree that this a bug, caused by strptime accepting non-zero padded values. I don't think there is an easy fix for this as other combinations of not-separated locale settings would than probably be interpreted wrongly.

What we really should do, is warn users in the documentation.

If anyone wants to have a closer look at this, please do!
